hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java

```diff
@@ -271,28 +269,40 @@ public void handleQuery(@NonNull final Bytes requestBuffer, @NonNull final Buffe
                     final var queryFees = handler.computeFees(context).totalFee();
 
                     final var header = createResponseHeader(responseType, OK, queryFees);
-                    response = handler.createEmptyResponse(header);
+                    return handler.createEmptyResponse(header);
                 } else {
                     // 6.ii Find response
                     final var header = createResponseHeader(responseType, OK, 0L);
-                    response = handler.findResponse(context, header);
+                    return handler.findResponse(context, header);
                 }
             } catch (InsufficientBalanceException e) {
-                response = createErrorResponse(handler, responseType, e.responseCode(), e.getEstimatedFee());
+                return createErrorResponse(handler, responseType, e.responseCode(), e.getEstimatedFee());
             } catch (PreCheckException e) {
-                response = createErrorResponse(handler, responseType, e.responseCode(), 0L);
+                return createErrorResponse(handler, responseType, e.responseCode(), 0L);
             } catch (HandleException e) {
                 // Conceptually, this should never happen, because we should use PreCheckException only for queries
                 // But we catch it here to play it safe
-                response = createErrorResponse(handler, responseType, e.getStatus(), 0L);
+                return createErrorResponse(handler, responseType, e.getStatus(), 0L);
             } catch (Exception e) {
                 logger.error("Unexpected exception while handling a query", e);
-                response = createErrorResponse(handler, responseType, FAIL_INVALID, 0L);
+                return createErrorResponse(handler, responseType, FAIL_INVALID, 0L);
             }
-        } else {
-            response = DEFAULT_UNSUPPORTED_RESPONSE;
+        } catch (PreCheckException e) {
+            return Response.newBuilder()
+                    .transactionGetReceipt(TransactionGetReceiptResponse.newBuilder()
```

Comment:
I do not think this is the right way to handle this situation. Somebody sends a query of a specific type and gets a response for a different kind of query. I doubt any client would have code that interprets the response correctly.

Reply:
I agree that TRANSACTION_HAS_UNKNOWN_FIELDS is not the best response code for failed queries, but I couldn't find anything better. Should I create a new response code QUERY_HAS_UNKNOWN_FIELDS?

Reply:
I am less worried about `TRANSACTION_HAS_UNKNOWN_FIELDS` but more so about the usage of `TransactionGetReceiptResponse`. Unless the client happens to send a `TransactionGetReceiptQuery`, it is unlikely to understand the response.

Reply:
As we discussed, I updated to throw a grpc error StatusRuntimeException:INVALID_ARGUMENT when either an invalid query parameter is set or a query name is invalid.